model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -16,89 +16,58 @@
-        ctc_greedy_decoder = match['decoder']
+        ctc_greedy_decoder_tf = match['decoder']
         cast = match['cast']
         sparse_to_dense = match['sparse_to_dense']
-        sparse_to_dense_name = sparse_to_dense.soft_get('name', sparse_to_dense.id)
-
-        # disconnect SparseToDense and Cast nodes
-        sparse_to_dense.in_port(0).disconnect()
-        cast.in_port(0).disconnect()
-
-        # transform CTCGreedyDecoder output to TensorFlow's one:
-        # 1. squeeze the output to [N, T] shape
-        # 2. cast it to integer
-        squeeze_dec_seq = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([2, 3])},
-                                                      {'name': sparse_to_dense_name})
-        squeeze_dec_seq.in_port(0).connect(ctc_greedy_decoder.out_port(0))
-        cast_to_int = Cast(graph, {'name': sparse_to_dense_name + '/CastToInt',
-                                   'dst_type': np.int32}).create_node()
-        cast_to_int.in_port(0).connect(squeeze_dec_seq.out_port(0))
 
-        # preserve output name from original graph
-        rename_nodes([(sparse_to_dense, sparse_to_dense_name + '/AbandonedName'),
-                      (cast_to_int, sparse_to_dense_name)])
+        # for normalizing input chanel need to transpose input data from [T, N, C] to [N, T, C]
+        # which supported CTCGreedyDecoderSeqLen op.
+        ctc_data_permute = create_op_with_const_inputs(graph, Transpose, {1: int64_array([1, 0, 2])},
+                                                            {'name': ctc_greedy_decoder_tf.name + '/ctc_data_permute'})
+        ctc_greedy_decoder_tf.in_port(0).get_source().connect(ctc_data_permute.in_port(0))
+        merge_repeated_tf = ctc_greedy_decoder_tf.soft_get('merge_repeated', ctc_greedy_decoder_tf.id)
+        ctc_greedy_decoder = CTCGreedyDecoderSeqLenOp(graph, {'name': ctc_greedy_decoder_tf.name,
+                                                              'cmerge_repeated': merge_repeated_tf}).create_node()
+        ctc_greedy_decoder.in_port(0).connect(ctc_data_permute.out_port(0))
+        ctc_greedy_decoder_tf.in_port(1).get_source().connect(ctc_greedy_decoder.in_port(1))
 
         # set output of the new sub-graph as a source for SparseToDense consumer
-        sparse_to_dense.out_port(0).get_connection().set_source(cast_to_int.out_port(0))
+        sparse_to_dense.out_port(0).get_connection().set_source(ctc_greedy_decoder.out_port(0))
 
         # remove no longer needed nodes
-        graph.remove_nodes_from([sparse_to_dense.id, cast.id])
-
-        # mark CTCGreedyDecoder node as a node that requires transformation of sequence length to a mask format
-        # in the middle phase
-        ctc_greedy_decoder['use_mask_format'] = True
+        graph.remove_nodes_from([sparse_to_dense.id, cast.id, ctc_greedy_decoder_tf.id])
 
         # unless the second input of CTCGreedyDecoder is a parameter, it enforces MO to use --static-shape
```

Comment:
I think that this code is not needed anymore. @rkazants , is that right?

Reply:
Exactly. It needs to remove. Thanks